lib/KalturaFTPServer.js

```diff
@@ -146,28 +160,30 @@ FtpSession.prototype = {
 					}
 				}
 				else {
-					this.getFileAndInsertToCache(callback, extension, file, service, id);
+					this.getFileAndInsertToCache(callback, this, extension, file, service, id);
 				}
 			}
 		}
 	},
 
-	handleXmlListResponse: function (listResponse, callback, This, service, currentList, map, pager, currentClient, handler, filter) {
+	handleXmlListResponse: function (listResponse, callback, This, service, currentList, map, pager, currentClient, handler, filter, retrieveOnePageOnly) {
```

Comment:
Don't pass This to the method.
Also, instead of passing retrieveOnePageOnly, you can tell that if the pager is not null then only one page should be retrieved.